flang/lib/Optimizer/CodeGen/TargetRewrite.cpp

```diff
@@ -168,13 +168,13 @@ class TargetRewrite : public TargetRewriteBase<TargetRewrite> {
     } else {
       assert(m.size() == 2);
       // COMPLEX is split into 2 separate arguments
-      auto iTy = rewriter->getIntegerType(32);
       for (auto e : llvm::enumerate(m)) {
         auto &tup = e.value();
         auto ty = std::get<mlir::Type>(tup);
         auto index = e.index();
-        auto idx = rewriter->getIntegerAttr(iTy, index);
-        auto val = rewriter->create<ExtractValueOp>(loc, ty, oper, rewriter->getArrayAttr(idx));
+        auto idx = rewriter->getIntegerAttr(rewriter->getIndexType(), index);
+        auto val = rewriter->create<ExtractValueOp>(
+            loc, ty, oper, rewriter->getArrayAttr(idx));
```

Comment:
Doesn't this switch from `i32` to `index` cause LLVM to assert and fail?

Reply:
It doesn't fails on fir-dev tests... Maybe because we stay in FIR and we do not lower to LLVM IR in the tests.

Reply:
Yes, that is what I was wondering/thinking.

Reply:
I'll have to double check that.

Reply:
So after a while I came back to this and you are right @schweitzpgi. I made a patch to revert that upstream and I'll update it here as well. 

https://reviews.llvm.org/D119253